api/taskgroup.go

```diff
@@ -0,0 +1,363 @@
+	BaseHandler
+	BucketHandler
+}
+
+//
+// AddRoutes adds routes.
+func (h TaskGroupHandler) AddRoutes(e *gin.Engine) {
+	routeGroup := e.Group("/")
+	routeGroup.Use(auth.AuthorizationRequired(h.AuthProvider, "tasks"))
+	routeGroup.GET(TaskGroupsRoot, h.List)
+	routeGroup.GET(TaskGroupsRoot+"/", h.List)
+	routeGroup.POST(TaskGroupsRoot, h.Create)
+	routeGroup.PUT(TaskGroupRoot, h.Update)
+	routeGroup.GET(TaskGroupRoot, h.Get)
+	routeGroup.PUT(TaskGroupSubmitRoot, h.Submit)
+	routeGroup.GET(TaskGroupBucketRoot, h.Content)
+	routeGroup.POST(TaskGroupBucketRoot, h.Upload)
+	routeGroup.PUT(TaskGroupBucketRoot, h.Upload)
+	routeGroup.DELETE(TaskGroupRoot, h.Delete)
+}
+
+// Get godoc
+// @summary Get a task group by ID.
+// @description Get a task group by ID.
+// @tags get
+// @produce json
+// @success 200 {object} api.TaskGroup
+// @router /taskgroups/{id} [get]
+// @param id path string true "TaskGroup ID"
+func (h TaskGroupHandler) Get(ctx *gin.Context) {
+	m := &model.TaskGroup{}
+	id := h.pk(ctx)
+	db := h.preLoad(
+		h.DB,
+		"Tasks")
+	result := db.First(m, id)
+	if result.Error != nil {
+		h.getFailed(ctx, result.Error)
+		return
+	}
+	r := TaskGroup{}
+	r.With(m)
+
+	ctx.JSON(http.StatusOK, r)
+}
+
+// List godoc
+// @summary List all task groups.
+// @description List all task groups.
+// @tags get
+// @produce json
+// @success 200 {object} []api.TaskGroup
+// @router /taskgroups [get]
+func (h TaskGroupHandler) List(ctx *gin.Context) {
+	var list []model.TaskGroup
+	db := h.preLoad(
+		h.DB,
+		"Tasks")
+	result := db.Find(&list)
+	if result.Error != nil {
+		h.listFailed(ctx, result.Error)
+		return
+	}
+	resources := []TaskGroup{}
+	for i := range list {
+		r := TaskGroup{}
+		r.With(&list[i])
+		resources = append(resources, r)
+	}
+
+	ctx.JSON(http.StatusOK, resources)
+}
+
+// Create godoc
+// @summary Create a task group.
+// @description Create a task group.
+// @tags create
+// @accept json
+// @produce json
+// @success 201 {object} api.TaskGroup
+// @router /taskgroups [post]
+// @param taskgroup body api.TaskGroup true "TaskGroup data"
+func (h TaskGroupHandler) Create(ctx *gin.Context) {
+	group := &TaskGroup{}
+	err := ctx.BindJSON(group)
+	if err != nil {
+		h.createFailed(ctx, err)
+		return
+	}
+	m := group.Model()
+	result := h.DB.Create(&m)
+	if result.Error != nil {
+		h.createFailed(ctx, result.Error)
+		return
+	}
+
+	group.With(m)
+
+	ctx.JSON(http.StatusCreated, group)
+}
+
+// Update godoc
+// @summary Update a task group.
+// @description Update a task group.
+// @tags update
+// @accept json
+// @success 204
+// @router /tasks/{id} [put]
+// @param id path string true "Task ID"
+// @param task body Task true "Task data"
+func (h TaskGroupHandler) Update(ctx *gin.Context) {
+	id := h.pk(ctx)
+	r := &TaskGroup{}
+	err := ctx.BindJSON(r)
+	if err != nil {
+		return
+	}
+	g := r.Model()
+	g.ID = id
+	err = h.DB.Transaction(
+		func(tx *gorm.DB) (err error) {
+			db := tx.Model(g)
+			db = db.Omit("Tasks")
```

Comment:
You can use `clause.Associations` in a call to Omit() rather than having to name each association. We can also probably use that to preload associations in most cases.

Reply:
So db.Omit(clauses.Associations)?

Reply:
Yep, `db = db.Omit(clause.Associations)`

Reply:
GoLand should automatically figure out to import `gorm.io/gorm/clause`